### PR TITLE
Remote Reindex UI improvements

### PR DESCRIPTION
**Pull request by gally47**
In this PR we:

- [x] improved the indices sort
- [x] added support for hostnames with / at the end
- [x] made sure the form value is saved when we retry to migrate

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by todvora — APPROVED** [review]
Tested locally and everything works as expected. One more detail - could we make the hostname mandatory? 
 
image

Thx!

**Review by ousmaneo — APPROVED** [review]
LGTM !

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx [hunk 1]
```diff
@@ -253,12 +273,10 @@ const MigrateExistingData = ({ currentStep, onTriggerStep, hideActions }: Migrat
             <Spinner />
           ) : (
             <MigrationStepTriggerButtonToolbar hidden={hideActions}
+                                               disabled={Object.values(errors).length > 0}
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/MigrateExistingData.tsx:276]: We can use `isValidating` and `isValid` from formik to disable the button

**Review by ousmaneo — CHANGES_REQUESTED** [review]
Looks good, I added a small comments

**Review by ousmaneo — APPROVED** [review]

**Pull request merged**